### PR TITLE
Fix - filter out common test filename patterns when requiring models

### PR DESCRIPTION
**Pull request by jdbdnz**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

## Issue
Error thrown on initialisation `ReferenceError: beforeAll is not defined`. `beforeAll` is a Jest function in a file with a path of `db/models/__tests__/*.spec.js`. The stacktrace shows it being required in `requireAllModels`.

## Expected behavior
Forest does not require test files and no error is thrown on initialisation related to Jest functions

## Steps to reproduce:
- Follow the convention of keeping test files with your source files; specifically have one in models directory
- Have that test file call a Jest function such as `beforeAll`
- Observe error thrown when test file is required by Forest during initialisation when models are required

### CODE REVIEW

**Comment by slimee** [comment]
[P3 - External - forest-express - Do not import test files in requireAllModels](https://trello.com/c/RoAgGNph/2172-p3-external-forest-express-do-not-import-test-files-in-requireallmodels)

File: src/index.js [hunk 1]
```diff
@@ -51,10 +51,11 @@ function getModels() {
 function requireAllModels(modelsDir) {
   if (modelsDir) {
     try {
+      const isJSFilename = (fileName) => fileName.endsWith('.js') || (fileName.endsWith('.ts') && !fileName.endsWith('.d.ts'));
+      const isTestFilename = (fileName) => fileName.match(/__tests__|.test.|.spec./g);
       requireAll({
         dirname: modelsDir,
```
Comment (rap2hpoutre) [inline src/index.js]: Exclude all directories that contains `__tests__`:

```suggestion
         dirname: modelsDir,
         excludeDirs: /^__tests__$/,
```

See: https://github.com/felixge/node-require-all#usage

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ## [Unreleased]
+### Fixed
+- Initialisation - Do not require files that have `__tests__`, `.test.`, or `.spec.` in their filename.
```
Comment (rap2hpoutre) [inline CHANGELOG.md]:
```suggestion
- Initialization - Filter out test files when requiring models (`__tests__/*`, `*.spec.js`, `*.spec.ts`, `*.test.js` or `*.test.ts`).
```

File: src/index.js [hunk 3]
```diff
@@ -51,10 +51,11 @@ function getModels() {
 function requireAllModels(modelsDir) {
   if (modelsDir) {
     try {
+      const isJSFilename = (fileName) => fileName.endsWith('.js') || (fileName.endsWith('.ts') && !fileName.endsWith('.d.ts'));
+      const isTestFilename = (fileName) => fileName.match(/__tests__|.test.|.spec./g);
```
[the remark this answers is not in the document]
Comment (jdbdnz) [reply src/index.js]: Oops! My intention was to only match test and spec when surrounded by periods. I think that expression should actually be `/__tests__|\.test\.|\.spec\./g`

As for `__tests__` being controlled for elsewhere. Could you link that code? Currently forest (forest-express@v5.4.1) picks up my path `/db/models/__tests__/content.spec.js`

**Comment by jdbdnz** [comment]
Thank you @rap2hpoutre for the thoughtful feedback. I've implemented your suggested changes.

**Comment by rap2hpoutre** [comment]
🎉 Thank you for your contribution @jdbdnz !

**Pull request merged**